onnxruntime/core/providers/cpu/generator/range.cc

```diff
@@ -80,24 +81,43 @@ static Status ComputeRange(OpKernelContext* ctx) {
   return Status::OK();
 }
 
+namespace range_internal {
+
+// Workaround GCC bug https://gcc.gnu.org/bugzilla/show_bug.cgi?id=47226
+// can not specify lambda directly
+template <class T, class... Args>
+inline int invoke_computerange_callable(size_t& called, Status& s, int32_t dt_type, Args&&... args) {
+  auto fn = [&] {
+    if (utils::ToTensorDataType<T>() == dt_type) {
+      s = ComputeRange<T>(std::forward<Args>(args)...);
+      ++called;
+    }
+    return 0;
+  };
+  return fn();
+}
+
+
+template<typename... Types>
+inline Status CallDispatcher(int32_t dt_type, OpKernelContext* ctx) {
+  Status s;
```

Comment:
Is it possible to generalize the dispatcher so we can use a common implementation everywhere?

Reply:
The problem I am facing is that we can't use templates in lambdas, especially with the current compiler.

---
In reply to: [341920846](https://github.com/microsoft/onnxruntime/pull/2307#discussion_r341920846) [](ancestors = 341920846)